modules/outbound-command-event-handler/test/integration/application/process_sdk_outbound_bulk_quotes_request.test.ts

```diff
@@ -0,0 +1,327 @@
+              partyIdentifier: "16135551212"
+            },
+          },
+          amountType: "SEND",
+          currency: "USD",
+          amount: "123.45",
+        },
+        {
+          homeTransactionId: randomUUID(),
+          to: {
+            partyIdInfo: {
+              partyIdType: "MSISDN",
+              partyIdentifier: "16135551213"
+            },
+          },
+          amountType: "SEND",
+          currency: "USD",
+          amount: "456.78",
+        }
+      ]
+    }
+    const sampleCommandEventData: IProcessSDKOutboundBulkRequestCmdEvtData = {
+      bulkRequest,
+      timestamp: Date.now(),
+      headers: []
+    }
+    const processSDKOutboundBulkRequestMessageObj = new ProcessSDKOutboundBulkRequestCmdEvt(sampleCommandEventData);
+    await producer.sendCommandEvent(processSDKOutboundBulkRequestMessageObj);
+    await new Promise(resolve => setTimeout(resolve, messageTimeout));
+
+    const bulkPartyInfoRequestCommandEventData: IProcessSDKOutboundBulkPartyInfoRequestCmdEvtData = {
+      bulkId: bulkTransactionId,
+      timestamp: Date.now(),
+      headers: []
+    }
+    const bulkPartyInfoRequestCommandEventObj = new ProcessSDKOutboundBulkPartyInfoRequestCmdEvt(
+      bulkPartyInfoRequestCommandEventData
+    );
+    await producer.sendCommandEvent(bulkPartyInfoRequestCommandEventObj);
+    await new Promise(resolve => setTimeout(resolve, messageTimeout));
+
+    // Check that the command handler sends 2 messages requesting party info.
+    const partyInfoRequestedDomainEvents = domainEvents.filter(domainEvent => domainEvent.getName() === 'PartyInfoRequestedDmEvt');
+    expect(partyInfoRequestedDomainEvents.length).toEqual(bulkRequest.individualTransfers.length);
+
+    // Get the randomly generated transferIds for the callback
+    const randomGeneratedTransferIds = await bulkTransactionEntityRepo.getAllIndividualTransferIds(bulkTransactionId);
+
+    // Simulate the domain handler sending the command handler PProcessPartyInfoCallback messages
+    // for each individual transfer
+    const processPartyInfoCallbackMessageData1: IProcessPartyInfoCallbackCmdEvtData = {
+      bulkId: bulkTransactionId,
+      content: {
+        transferId: randomGeneratedTransferIds[0],
+        partyResult: {
+          party: {
+              partyIdInfo: {
+                  partyIdType: 'MSISDN',
+                  partyIdentifier: '123456',
+                  fspId: 'receiverfsp'
+              }
+          },
+          currentState: 'COMPLETED'
+        },
+      },
+      timestamp: Date.now(),
+      headers: []
+    }
+    const processPartyInfoCallbackMessageData2: IProcessPartyInfoCallbackCmdEvtData = {
+      bulkId: bulkTransactionId,
+      content: {
+        transferId: randomGeneratedTransferIds[1],
+        partyResult: {
+          party: {
+              partyIdInfo: {
+                  partyIdType: 'MSISDN',
+                  partyIdentifier: '123456',
+                  fspId: 'receiverfsp'
```

Comment:
Can we have a different receiver fspId to make sure this ends up in separate quote batch.

Reply:
Done. Added a lookup for `differentfsp` and I assert that it is in a batch by itself.